src/pynwb/base.py

```diff
@@ -282,3 +285,107 @@ def __init__(self, **kwargs):
+                        "to be selected as well as an object reference to the TimeSeries."},
+            *get_docval(VectorData.__init__, 'data'))
+    def __init__(self, **kwargs):
+        call_docval_func(super().__init__, kwargs)
+
+    def get(self, key, **kwargs):
+        """
+        Retrieve elements from this object.
+
+        The function uses :py:class:`~pynwb.base.TimeSeriesReferenceVectorData.TIME_SERIES_REFERENCE_TUPLE`
+        to describe individual records in the dataset. This allows the code to avoid exposing internal
+        details of the schema to the user and simplifies handling of missing values by explictly
+        representing missing values via
+        :py:class:`~pynwb.base.TimeSeriesReferenceVectorData.TIME_SERIES_REFERENCE_NONE_TYPE`
+        rather than the internal representation used for storage of ``(-1, -1, TimeSeries)``.
+
+        :param key: Selection of the elements
+        :param kwargs: Ignored
+
+        :returns: :py:class:`~pynwb.base.TimeSeriesReferenceVectorData.TIME_SERIES_REFERENCE_TUPLE` if a single
+                  element is being selected. Otherwise return a list of
+                  :py:class:`~pynwb.base.TimeSeriesReferenceVectorData.TIME_SERIES_REFERENCE_TUPLE` objects.
+                  Missing values are represented by
+                  :py:class:`~pynwb.base.TimeSeriesReferenceVectorData.TIME_SERIES_REFERENCE_NONE_TYPE`
+                  in which all values (i.e., idx_start, count, timeseries) are set to None.
+        """
+        vals = super().get(key)
+        # we only selected one row.
+        if isinstance(key, (int, np.integer)):
+            # NOTE: If we never wrote the data to disk, then vals will be a single tuple.
+            #       If the data is loaded from an h5py.Dataset then vals will be a single
+            #       np.void object. I.e., an alternative check would be
+            #       if isinstance(vals, tuple) or isinstance(vals, np.void):
+            #          ...
+            if vals[0] < 0 or vals[1] < 0:
+                return self.TIME_SERIES_REFERENCE_NONE_TYPE
+            else:
+                return self.TIME_SERIES_REFERENCE_TUPLE(*vals)
+        else:  # key selected multiple rows
+            # When loading from HDF5 we get an np.ndarray otherwise we get list-of-list. This
+            # makes the values consistent and tranforms the data to use our namedtuple type
+            re = [self.TIME_SERIES_REFERENCE_NONE_TYPE
+                  if (v[0] < 0 or v[1] < 0) else self.TIME_SERIES_REFERENCE_TUPLE(*v)
+                  for v in vals]
+            return re
+
+    def extend(self, ar):
+        """
+        The extend_data method adds all the elements of the iterable arg to the
+        end of the data of this Data container.
+
+        :param arg: The iterable to add to the end of this VectorData
+        """
+        self.data = extend_data(self.data, ar)
```

Comment:
Is this necessary? Also it doesn't work. 

```python
>>> v = pynwb.base.TimeSeriesReferenceVectorData(name='a', description='a')
>>> v.extend([1])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\users\ryan\documents\nwb\pynwb\src\pynwb\base.py", line 391, in extend
    self.data = extend_data(self.data, ar)
AttributeError: can't set attribute
```

Reply:
Removing the code and using `VectorData.extend` seems to work just fine.

Reply:
Fixed in https://github.com/NeurodataWithoutBorders/pynwb/pull/1349/commits/a58921aca2401c9c809b00a7bec47bf30d977558